app/controllers/adhoq/executions_controller.rb

```diff
@@ -44,7 +44,17 @@ def async_execution?
     end
 
     def query_parameters
-      params[:parameters] || HashWithIndifferentAccess.new
+      if params[:parameters]
+        if params[:parameters].instance_of?(Hash)
+          params[:parameters]
+        else
+          # for after Rails5
+          params[:parameters].permit!
+          params[:parameters].to_h
```

Comment:
Can't these two lines be written this?

``` ruby
params[:paremeters].to_unsafe_hash
```

(Only this condition has returned pure Hash, but is that OK?)